lldb/source/Commands/CommandObjectPlatform.cpp

```diff
@@ -180,7 +184,19 @@ class CommandObjectPlatformSelect : public CommandObjectParsed {
             m_interpreter, ArchSpec(), select, error, platform_arch));
         if (platform_sp) {
           GetDebugger().GetPlatformList().SetSelectedPlatform(platform_sp);
-
+          OptionGroupPythonClassWithDict &script_class_opts =
```

Comment:
What happens if one tries to set metadata on a platform that doesn't support/care about it?

Reply:
> Nothing.

Okay. Is that what we want to happen? Should we maybe make that an error, or at least give the user a warning that the arguments it laboriously typed are going to be ignored?

Reply:
So `Platform::SetMetadata` returns void, but `Platform::ReloadMetadata` returns an `llvm::Error` as you can see down below (line 196). When setting it from the command line, the user will see an error message and the command interpreter however from the SBAPI, since this is called from the `SBPlatform` constructor, there is no way to do error reporting (without adding an overload for the constructor), so the error is consumed and logged to the `lldb platform` channel.

Reply:
Okay, this is better, though a more user-focused error message would be even better ("platform XXX does not support scripting metadata" ?) I'm less worried about what happens in the SB API version, though if the operation can fail for more than one reason, then it might be a good idea to add an `SBError&` to the appropriate constructor.

Reply:
Or don't specify the metadata on the constructor, but with an accessor that returns a SBError and omit it from the constuctor.

Reply:
@clayborg I considered that but that would mean that the platform is in an invalid state after construction and it would require the user to call that accessor in order to "complete" its construction. I think adding an overload with an extra `SBError&`  would be fine here.